widget-src/components/ChangeLogRow.tsx

```diff
@@ -62,7 +69,44 @@ export const ChangeLogRow = ({
             width="fill-parent"
             verticalAlignItems="center"
           >
-            {/* <Type type="Added" /> */}
+            {!!changeLog.showTypeMenu && (
+              <TypeMenu
+                currentType={changeLog.type === 'added' ? 'none' : changeLog.type}
+                selectType={(newType) => {
+
+                  const addEdit = changeLog.type !== 'none' && changeLog.type !== 'added';
+
+                  if (newType !== changeLog.type) {
+                      updateChange({
+                        type: newType,
+                        editCount: addEdit ? changeLog.editCount + 1 : changeLog.editCount,
+                        editedDate: addEdit ? Date.now() : changeLog.editedDate,
+                        showTypeMenu: !changeLog.showTypeMenu,
+                      });
+                      setUpdatedDate(Date.now());
+                  } else {
+                    updateChange({
+                      showTypeMenu: !changeLog.showTypeMenu,
+                    })
+                  }
+                }}
+              />
+            )}
+            {showTypes && (
+              <Type
+                type={changeLog.type} 
+                action={() => {
+                  // toggle log type menu
+                  updateChange({
+                    showTypeMenu: !changeLog.showTypeMenu,
+                  })
+                  // hide all other log type menues
```

Comment:
👏 I didn't think we were able to do this without perf issues, thank you!

Reply:
It might have perf issues with a ton (~1000) logs since it loops through and sets state on each one, but there are probably other issues at that point anyhow.